doc/developer/design/20230306_persist_mfp_pushdown.md

```diff
@@ -0,0 +1,357 @@
+## Schemas
+
+Understanding schemas at the persistent data level gives us a building block [to explore schema evolution at
+the Persist level](https://github.com/MaterializeInc/materialize/issues/16625).
+
+## Internal Improvements
+
+Less tangible motivation for this work is that by columnarizing and summarizing parts, we will have a forcing 
+function to progress some lesser-seen concerns within Persist. Examples of this include how proper columnar 
+storage will reduce allocations and memory pressure caused by Persist (something we're already grappling with), 
+and how storing statistics will require a more long-term scalable solution to how we store metadata. These are
+problems we know we must stay on top of, and working through the building blocks for MFP pushdown will help frame
+a long-term vision to solving them.
+
+# Explanation
+[explanation]: #explanation
+
+A large amount of the changes needed for MFP pushdown are within Persist, where we must first build in primitives
+to understand and summarize the data we receive. The interface to this work relevant for other teams (Compute specifically)
+[is noted below in the Reference explanation](#filtering-parts).
+
+## Data Types
+
+Persist will need to understand individual columns of data, which will require newly supporting data types.
+These data types will be columnar encoded using Arrow/Parquet for their in-mem/durable representation,
+and allow for aggregations (e.g. min and max calculations) over columns.
+
+## Schemas
+
+A `Schema` trait will be introduced to map data back and forth between Persist columns. Any Rust type
+that is to roundtrip through Persist must have a corresponding `Schema`.
+
+## Part Statistics
+
+When Persist writes a batch to Blob storage, it will compute basic statistics over its columns, when possible,
+and write this data alongside existing metadata in Consensus. To start, we will collect a column's 
+non-null min and max values and null count, as well as the min/max Row within the part. Statistics will always 
+be optional, as Persist may not be able to compute min/max values for all types under all situations.
+
+## MFP
+
+### Filtering
+
+Before batch parts are fetched from Blob storage in `persist_source`, the dataflow's MFP is applied to 
+each part's statistics to determine whether it contains data relevant to the dataflow. If not, we get to
+skip fetching the part from Blob storage entirely.
+
+Note that:
+* A part may or may not have statistics for all of its columns
+* The MFP's filter will need to be modified to apply to min/max ranges, rather than individual rows
+* In the absence of sufficient statistics or an applicable filter, we default to fetching the part like
+we currently do. 
+
+### Projection
+
+Once a part is fetched, there is additional opportunity to push down our MFP's projection by only decoding
+the columns necessary for the dataflow. This is particularly valuable for datasets with wide rows that 
+typically see narrower reads.
```

Comment:
I'd say that eventually we can push the projection down to not filtering this column in the first place, but that's out of scope for this initial work.